azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/JoinCallButtonHolderViewModel.kt

```diff
@@ -20,25 +21,32 @@ import kotlinx.coroutines.flow.StateFlow
 
 internal class JoinCallButtonHolderViewModel(
     private val dispatch: (Action) -> Unit,
-    private val audioManager: AudioManager
+    private val audioManager: AudioManager,
+    private val isTelecomManagerEnabled: Boolean = false
 ) {
 
     private lateinit var joinCallButtonEnabledFlow: MutableStateFlow<Boolean>
+    private lateinit var callingStatusStateFlow: MutableStateFlow<CallingStatus>
     private var disableJoinCallButtonFlow = MutableStateFlow(false)
     private lateinit var networkManager: NetworkManager
+    private var callType: CallType? = null
 
     fun getJoinCallButtonEnabledFlow(): StateFlow<Boolean> = joinCallButtonEnabledFlow
 
     fun getDisableJoinCallButtonFlow(): StateFlow<Boolean> = disableJoinCallButtonFlow
 
+    fun getCallingStatusStateFlow(): StateFlow<CallingStatus> = callingStatusStateFlow
+
+    fun getCallType(): CallType? = callType
+
     fun launchCallScreen() {
         val networkAvailable = isNetworkAvailable()
         // We try to check for mic availability for the current application through current audio mode
         val normalAudioMode = audioManager.mode == AudioManager.MODE_NORMAL
 
         if (!networkAvailable) {
             handleOffline()
-        } else if (!normalAudioMode) {
+        } else if (!normalAudioMode && !isTelecomManagerEnabled) {
```

Comment:
!(normalAudioMode || isTelecomManagerEnabled)

https://en.wikipedia.org/wiki/De_Morgan%27s_laws

I find the other form a bit easier to read, but it's really preference.

Reply:
As per discussion we will fix for GA.